MdePkg/Include/Register/Intel/StmApi.h

```diff
@@ -34,6 +36,16 @@ typedef struct {
 #define STM_SPEC_VERSION_MAJOR  1
 #define STM_SPEC_VERSION_MINOR  0
 
+// MU_CHANGE - [START]: Add new CPU information header for STM
+#if defined (_MSC_EXTENSIONS) || defined (__cplusplus)
+#define STATIC_ASSERT  static_assert
+#else
+#define STATIC_ASSERT  _Static_assert

```

Comment:
This is already defined in [Base.h](https://github.com/tianocore/edk2/blob/master/MdePkg/Include/Base.h#L795-L809).

Reply:
That's true.  The issue that arises is that when building the genstm tools in BaseTools it references this file but it never looks at Base.h.  This causes it to fail with STATIC_ASSERT.

Reply:
It's the same reason we need to include redefinitions of SIZE_1KB and SIZE_2KB in GenStm.h.  I can move this redefinition to there if you prefer

Reply:
In a separate commit so this will become rebase and fast-forward with the original change to STM_HEADER.